libraries/chain/fork_database.cpp

```diff
@@ -24,12 +24,12 @@ namespace eosio::chain {
 
    struct block_state_accessor {
       static bool is_valid(const block_state& bs) { return bs.is_valid(); }
-      static void set_valid(block_state& bs, bool v) { bs.validated.store(v); }
+      static void set_valid(block_state& bs, bool v) { bs.set_valid(v); }
    };
 
    struct block_state_legacy_accessor {
       static bool is_valid(const block_state_legacy& bs) { return bs.is_valid(); }
-      static void set_valid(block_state_legacy& bs, bool v) { bs.validated = v; }
+      static void set_valid(block_state_legacy& bs, bool v) { bs.set_valid(v); }
    };
```

Comment:
At this point, there is no difference between the two, so we don't need the `block_state_accessor` anymore I believe. 
For access we can just make the `fork_database_impl<>` friend in block_state and block_state_legacy, just like we do `controller_impl`.